### PR TITLE
ddl: Design doc of unified distributed parallel execution framework.

### LINKED ISSUE #43010: Design doc
Add phase I design doc for unified distributed parallel execution framework.

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: close #43010 

Problem Summary:
Add Design doc.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [X] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/design/2023-04-11-dist-task.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+```go
+// Dispatch defines the interface for operations inside a dispatcher.
+type Dispatch interface {
+// Start enables dispatching and monitoring mechanisms.
+Start()
+// GetAllSchedulerIDs gets handles the task's all available instances.
+GetAllSchedulerIDs(ctx context.Context, gTaskID int64) ([]string, error)
+// Stop stops the dispatcher.
+Stop()
+}
```
Comment (tangenta) [inline docs/design/2023-04-11-dist-task.md:101]: Please format these code snippets.

https://github.com/pingcap/tidb/blob/0eb5b3645787555ec7fb6862259bbb9ba31648d6/docs/design/2023-04-11-dist-task.md

File: docs/design/2023-04-11-dist-task.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
+# Proposal: Unified distributed parallel execution framework
+
+- Author: [Benjamin](https://github.com/Benjamin2037)
+- Tracking Issue:https://github.com/pingcap/tidb/issues/41495
+
+## Background and benefits
+TiDB is a distributed database that provides high scalability, available and performance for database service. Starting from version v7.1,
```
Comment (wjhuang2016) [inline docs/design/2023-04-11-dist-task.md:7]: Design doc should not mention version-related stuff.

**Review by wjhuang2016 — APPROVED** [review]

File: docs/design/2023-04-11-dist-task.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+6. Resource pool: For the dispatcher, scheduler, and executor, we limit them into a pool according to class. a global resource manager take care of limit the framework runtime resource usage quantity.
+
+## Detail Design
+
+This section introduces the data structures and interfaces that define the detail design of the framework.
+
+### System Variable
+
+A system variable is used to enable/disable the framework.
+`tidb_enable_dist_task`
```
Comment (tangenta) [inline docs/design/2023-04-11-dist-task.md:56]:
```suggestion
The system variable `tidb_enable_dist_task` is used to enable/disable the framework.

```

File: docs/design/2023-04-11-dist-task.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+### System Variable
+
+A system variable is used to enable/disable the framework.
+`tidb_enable_dist_task` 
+```sql
+mysql> set global tidb_enable_dist_task = ON|OFF;
+```
+### Data Structure
+In implementation, the following tables are used to represent the global task queue and global subtask queue:
+`Global task queue`
```
Comment (tangenta) [inline docs/design/2023-04-11-dist-task.md:62]:
```suggestion

### Data Structure

In implementation, the following two tables are used to represent the global task queue and global subtask queue:

```

File: docs/design/2023-04-11-dist-task.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+| dispatcher_id     | VARCHAR(256)                                   |      |
+| state             | VARCHAR(64) NOT NULL                           |      |
+| start_time        | TIMESTAMP                                      |      |
+| state_update_time | TIMESTAMP                                      |      |
+| meta              | LONGBLOB                                       |      |
+| concurrency       | INT(11)                                        |      |
+| step              | INT(11)                                        |      |
+
+`Global subtask queue`
+
```
Comment (tangenta) [inline docs/design/2023-04-11-dist-task.md:78]:
```suggestion
```

File: docs/design/2023-04-11-dist-task.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+	wg        sync.WaitGroup
+	logCtx    context.Context
+
+	mu struct {
+		sync.RWMutex
+		err error
+		// runtimeCancel is used to cancel the Run/Rollback when error occurs.
+		runtimeCancel context.CancelFunc
+	}
+}
```
Comment (tangenta) [inline docs/design/2023-04-11-dist-task.md:158]: Maybe we don't need to mention the implementation details here.

File: docs/design/2023-04-11-dist-task.md (hunk 5 above)
Comment (Benjamin2037) [reply docs/design/2023-04-11-dist-task.md:78]: Why delete this line？

File: docs/design/2023-04-11-dist-task.md (hunk 6 above)
Comment (Benjamin2037) [reply docs/design/2023-04-11-dist-task.md:158]: Only interface definition.

**Review by tangenta — APPROVED** [review]

File: docs/design/2023-04-11-dist-task.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
+# Proposal: Unified Distributed Parallel Execution Framework
+
+- Author: [Benjamin](https://github.com/Benjamin2037)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/41495
+
+## Background and Benefits
+
+
```
Comment (Benjamin2037) [inline docs/design/2023-04-11-dist-task.md:8]:
```suggestion
```

**Pull request merged**